### ISSUE TITLE
edgetpu produces quite different results from tflite on CPU.

### ISSUE THREAD

**Issue by YijinLiu** [google-coral/edgetpu#347]
I am using edgetpu to run a SSD2 model. I found that quite frequently (maybe 20% of the time) edgetpu produces very different results (mostly it's missed detection) from running the same model using tflite on a CPU.

This is the code I use to convert to edgetpu model:
<pre>
    converter = lite.TFLiteConverter.from_frozen_graph(
        flags.data_dir + "/exported/tflite_graph.pb",
        ["normalized_input_image_tensor"],
        ["TFLite_Detection_PostProcess",
         "TFLite_Detection_PostProcess:1",
         "TFLite_Detection_PostProcess:2",
         "TFLite_Detection_PostProcess:3"],
        {"normalized_input_image_tensor": [1, flags.height, flags.width, 3]})
    converter.allow_custom_ops = True
    converter.change_concat_input_ranges = False
    converter.optimizations = [lite.Optimize.DEFAULT]
    converter.output_format = lite_constants.TFLITE
    converter.quantized_input_stats = {"normalized_input_image_tensor": [128, 127.0]}
    def _representative_data_gen():
        for example in tfrecord.parse([flags.data_dir + '/train.tfrecord-00000-of-00100']):
            encoded = example["image/encoded"].numpy()
            image = tf.image.decode_jpeg(encoded)
            image = tf.image.resize(image, [flags.width, flags.height])
            image = tf.cast(image, tf.float32) / 255.0 - 0.5 
            image = tf.expand_dims(image, 0)
            yield [image]
    converter.representative_dataset = _representative_data_gen
    tflite_model_quant = converter.convert()
</pre>

Is there anything I did wrongly?

**Comment by Namburger** [comment]
@YijinLiu that's odd, could you share the model? I can try  running through our internal compiler to see if you can achieve better results

**Comment by YijinLiu** [comment]
You can find the model here: https://drive.google.com/drive/folders/1j-SAnqmZc7Wg5HKo80SoO6a9UXeQPNQ_?usp=sharing
tflite_graph.pb is the exported tflite graph using research/object_detection/export_tflite_ssd_graph.py

I used the following command to convert it to tflite model detect.tflite
<pre>
tflite_convert --graph_def_file=tflite_graph.pb --output_file=detect.tflite --input_shapes=1,300,300,3 \
               --input_arrays=normalized_input_image_tensor  \
               --output_arrays='TFLite_Detection_PostProcess','TFLite_Detection_PostProcess:1','TFLite_Detection_PostProcess:2','TFLite_Detection_PostProcess:3' \
               --inference_type=FLOAT --allow_custom_ops
</pre>

You already saw the script to convert tflite_graph.pb to detect_uint8.tflite in the first post. I used the following command to convert it to detect_uint8_edgetpu.tflite
<pre>
edgetpu_compiler --out_dir=. detect_uint8.tflite
</pre>

The tensorflow version is 1.15.4.

Thanks!

**Comment by YijinLiu** [comment]
I also added a test image in the same drive folder.
To make it clear: 
- detect.tflite works fine
- detect_uint8.tflite and detect_uint8_edgetpu.tflite have the same issue.

**Comment by Namburger** [comment]
@YijinLiu wait, so this is a matter of post training quantization?
The way I understand it, the non integer quantized version works but the quantized version is flaky?

**Comment by YijinLiu** [comment]
Right. I just realized that it might be the quantization issue when I read this [issue](https://github.com/google-coral/edgetpu/issues/191) and did a test to confirm that. Sorry that I didn't make it clear initially.
Is this something I should expect?

**Comment by ricardodeazambuja** [comment]
Hi @Namburger. 

I'm seeing a big degradation in the performance from tflite to an edgetpu compiled model. I trained a mask/no-mask/poor-mask model based on [the notebook you shared (btw, thanks for sharing!)](https://github.com/Namburger/edgetpu-ssdlite-mobiledet-retrain/blob/master/mask_detection.ipynb) and I only changed minor things like increasing the data augmentation. When I test the "pure" tflite model, the model works pretty well. However, the compiled one is close to rubbish. The same image that gets a nice bbox with 0.8 score, returns a poor bbox and score around 0.4 when using the edgetpu.

By reading the config file, I understood the model was trained with "quantization aware training". Here are the tflite and the compiled files: [ssdlite_mobiledet_mask.zip](https://github.com/google-coral/edgetpu/files/6443144/ssdlite_mobiledet_mask.zip)

Is there anything that could help improving the results in the final compiled model?
